test/functional/feature_proxy.py

```diff
@@ -60,14 +63,17 @@
 # Networks returned by RPC getnetworkinfo, defined in src/rpc/net.cpp::GetNetworksInfo()
 NETWORKS = frozenset({NET_IPV4, NET_IPV6, NET_ONION, NET_I2P, NET_CJDNS})
 
+# Use the shortest temp path possible since socket paths have a 92-char limit
+socket_path = os.path.join(tempfile.gettempdir(), "sockpath")
```

Comment:
See this ci failure: https://cirrus-ci.com/task/5039125594636288?logs=ci#L2638-L2640

This is why I used a shorter path, sometimes TMPDIR is > 50 characters! You have a good point about getting max `sun_path` from the system though, maybe it will balance out

```
File "/private/var/folders/v7/fs2b0v3s0lz1n57gj9y4xb5m0000gn/T/cirrus-ci-build/ci/scratch/build/bitcoin-arm64-apple-darwin/test/functional/test_framework/socks5.py", line 131, in __init__
self.s.bind(conf.addr)
OSError: AF_UNIX path too long
```

Reply:
I see. Then, to avoid multiple tests colliding on `/tmp/socket`, maybe this:

```python
socket_path = os.path.join(tempfile.TemporaryDirectory(), "s")
```